backend/tournesol/views/preview.py

```diff
@@ -260,14 +313,339 @@ def get(self, request, uid):
+        )
+
+        self._draw_logo(preview_image, entity, upscale_ratio=upscale_ratio)
+
         preview_image.save(response, "png")
         return response
+
+
+class ComparisonPreviewGenerator:
+    def __init__(self):
+        # An upscale ratio > 1 generates a bigger image while keeping all the relative sizes
+        # and positions identical (like a zoom). Only integer values are allowed for now.
+        self.upscale_ratio = 2
+
+    def render(self, entity_a, entity_b, thumbnail_a, thumbnail_b):
+        # pylint: disable=too-many-locals
+
+        base_size = (440, 240)
+
+        # All the sizes and positions in this method are relative to the base size, not the
+        # upscaled size. The other methods are responsible for calculating the final (upscaled)
+        # sizes and positions.
+
+        base_margin = 6
+        horizontal_separator_height = base_margin
+        margin_above_entity_descriptions = base_margin
+        entity_descriptions_horizontal_margin = base_margin
+        logo_size = 32
+        margin_above_slider = base_margin * 3
+        margin_below_slider = base_margin
+        vs_band_width = 34
+        vs_tilt_in_pixels = 10
+        text_below_slider = "Which one should be largely recommended?"
+
+        final = self.generate_image(base_size, color=COLOR_WHITE_BACKGROUND)
+
+        thumbnail_size = self.draw_thumbnail(
+            thumbnail_a,
+            target=final,
+            position=(0, 0),
+            width=base_size[0] // 2,
+        )
+        thumbnail_size = self.draw_thumbnail(
+            thumbnail_b,
+            target=final,
+            position=(base_size[0] // 2, 0),
+            width=base_size[0] // 2,
+        )
+
+        self.draw_vs_background(
+            target=final,
+            band_width=vs_band_width,
+            tilt_in_pixels=vs_tilt_in_pixels,
+            height=thumbnail_size[1],
+        )
+        self.draw_horizontal_separator(
+            target=final,
+            top=thumbnail_size[1],
+            height=horizontal_separator_height,
+        )
+
+        self.draw_text(
+            target=final,
+            text="VS",
+            position=(base_size[0] // 2, thumbnail_size[1] // 2),
+            font=self.font(FOOTER_FONT_LOCATION, 20),
+            fill=COLOR_BROWN_FONT,
+            anchor="mm",
+        )
+
+        entity_description_top = (
+            thumbnail_size[1]
+            + horizontal_separator_height
+            + margin_above_entity_descriptions
+        )
+
+        self.draw_logo(
+            target=final,
+            size=logo_size,
+            position=(base_size[0] // 2 - logo_size // 2, entity_description_top),
+        )
+
+        available_width = (
+            base_size[0] // 2
+            - logo_size // 2
+            - entity_descriptions_horizontal_margin * 2
+        )
+        entity_description_height = self.draw_entity_description(
```

Comment:
Should we avoid an assignation here, if the name `entity_description_height` is re-assigned another value just after?

Reply:
I hesitated but I thought it was better to make the 2 instructions identical to make it clear we're doing almost the same thing twice.

It may be better to get the 2 values in different variables and use the the maximum of them. I didn't bother because they return the same value right now and it's likely we will change this section if the sizes ever differ in future changes.

But I agree it's weird and it makes you wonder about the validity of the code so I changed that in 23e421493818307919edeedf40e3b62828de6714.